### ISSUE TITLE
Use latest versions of app by default

### ISSUE THREAD

**Issue by paulmasson** [ladybug-tools/spider-covid-19-viz-3d#9]
Now that #7 has been merged (and will presumably make it into all dev versions in progress), how about reverting the index files to use GitHub API to pull file list from `app` folder and use appropriate latest versions for both stable and dev? Will save you effort going foward!

**Comment by theo-armour** [comment]
@paulmasson 

For sure!

Who to do? Me || you?

old index is in archive folder

**Comment by paulmasson** [comment]
@theo-armour taking a crack at it by copying your functions from app

**Comment by paulmasson** [comment]
@theo-armour new index available in #10 for your review. If acceptable I'll add the dev version as well.

Please not that I moved the `window.history.pushState()` line to the very end of the file. Since you use the current URL multiple times in the app, it's best not to reset the address bar until the last.

**Comment by paulmasson** [comment]
@theo-armour updated dev index available in #11

**Comment by theo-armour** [comment]
@paulmasson 

Is dev ok to go stable?

If so make it happen!

**Comment by paulmasson** [comment]
@theo-armour I am assuming that when you start a new version you consider the latest one to be stable. The most current version has bad z-fighting, so no I do not think it is ready for prime time.

This pull request merely points the index file in `dev` towards the last iteration in `app`.

**Comment by paulmasson** [comment]
@theo-armour I'm also waiting to see my two PRs on stable show up in the latest. Again the question of how things will be updated.

**Comment by theo-armour** [comment]
@paulmasson 

> you start a new version you consider the latest one to be stable.

yes, a a copy of it with new date versioned name is the new dev.
 
the z-fighting does not bother me that much - it will get fixed soon enough. And since it's a much more informative version I would put that up as stable as soon as possible if it were up to me. ;-)

**Comment by paulmasson** [comment]
@theo-armour it's your repository, so if you have a new dev going then commit it. It will actually make it easier for me to work on the z-fighting, because I can then just do it in the stable.

But once again, I'd like to see the two PRs in the current dev. If I add them, will you sync with your computer before starting a new dev or have you already done that?

**Comment by theo-armour** [comment]
@paulmasson 

I synced. And added a new new version.

seems to be working. I'm clicking on the links in the read me to check

**Comment by paulmasson** [comment]
OK, let's try this one more time:

1) You didn't add the two double quotes INSIDE the regex for Korea, so the double quotes still show up around the name in the tool tip.

2) You didn't add the `window.history.pushState()` line to the end of your dev version before starting a new one. What is now the current stable does not have a pretty link in the address bar as a result.

Both of these changes need to go into the two most recent iterations, and that needs to be done by you so that I know you are actually merging my commits. I'm not going to patch every single version you push. GOT IT?

**Comment by theo-armour** [comment]
@paulmasson 

> GOT IT?

fingers crossed

would it be a good idea for me to delete the pushState in older versions in the same folder?

**Comment by paulmasson** [comment]
> would it be a good idea for me to delete the pushState in older versions in the same folder?
Why would you? It just makes a pretty link. Doesn't affect any other functionality.

Now please fix the Korean regex in both versions!

**Comment by paulmasson** [comment]
@theo-armour the pushState is NOT in the current dev version!

**Comment by theo-armour** [comment]
@paulmasson 

by golly, you were right!

**Comment by theo-armour** [comment]
@paulmasson 

what do you think of having two folders "dev" and "stable"?

that might be a way of keeping some of my horse puckies out of the stable?

**Comment by paulmasson** [comment]
@theo-armour this is not about experimental stuff, it's about assuring that everyone's work is incorporated on an ongoing basis. You like working on a local version and then just pushing it to the repository, which means I can't do PRs against any file you touch without losing everything I do.

Your method of working is quite frankly not geared toward collaboration. For stable collaboration there should only be one version being updated at a time, with someone checking for any updates and merging them BEFORE rushing on to the next new feature. Having two folders won't change anything.

And `pushState()` is still not showing up in the current dev version...

**Comment by theo-armour** [comment]
@paulmasson 

> And pushState() is still not showing up in the current dev version...

that was intentional. I was not certain it whether is is wanted or needed in the dev version.

Not having it in dev means I can more easily verify that the file name and the meta version date are the same.

> Your method of working is quite frankly not geared toward collaboration.

I hear you loud and clear. The only thing my method does is enable to "Stay hungry. Be foolish".

I see the dev folder as being a place where I do my experimenting. Stable folder is yours. A file only goes from dev to stable when you decide it's OK and safe to do so.

 From time to time, I will let you know that I think there is a version ready for you to look at. at that time you bang on it and tell me what to do in order to get it ready to get to your folder.

as and when you are happy with the latest Dead Files then you copy it over to the stable folder.

After that any changes should only be minor or cosmetic.

if you do want to make some big changes then you have your own dev folder. and when you feel ready you asked me to bang on it.

**Comment by paulmasson** [comment]
@theo-armour will think about your last comments but still think an extra folder won't change the issue of continuous updates. Let's keep working as is for now.

Opened PR #12 but only added your name as an edit, so don't know if you've been notified yet.

Closing this issue as fixed.
